code/datums/holocall.dm

```diff
@@ -0,0 +1,159 @@
+			dialed_holopads += H
+			LAZYADD(H.holo_calls, src)
+
+	if(!dialed_holopads.len)
+		calling_holopad.atom_say("Connection failure.")
+		qdel(src)
+		return
+
+	log_debug("Holocall started")
+
+//cleans up ALL references :)
+/datum/holocall/Destroy()
+	QDEL_NULL(eye)
+
+	user.reset_perspective()
+
+	user = null
+	hologram.HC = null
+	hologram = null
+	calling_holopad.outgoing_call = null
+
+	for(var/I in dialed_holopads)
+		var/obj/machinery/hologram/holopad/H = I
+		LAZYREMOVE(H.holo_calls, src)
+	dialed_holopads.Cut()
+
+	if(calling_holopad)
+		calling_holopad.SetLightsAndPower()
+		calling_holopad = null
+	if(connected_holopad)
+		connected_holopad.SetLightsAndPower()
+		connected_holopad = null
+
+	log_debug("Holocall destroyed")
+
+	return ..()
+
+//Gracefully disconnects a holopad `H` from a call. Pads not in the call are ignored. Notifies participants of the disconnection
+/datum/holocall/proc/Disconnect(obj/machinery/hologram/holopad/H)
+	testing("Holocall disconnect")
+	if(H == connected_holopad)
+		calling_holopad.atom_say("[usr] disconnected.")
+	else if(H == calling_holopad && connected_holopad)
+		connected_holopad.atom_say("[usr] disconnected.")
+
+	ConnectionFailure(H, TRUE)
+
+//Forcefully disconnects a holopad `H` from a call. Pads not in the call are ignored.
+/datum/holocall/proc/ConnectionFailure(obj/machinery/hologram/holopad/H, graceful = FALSE)
+	testing("Holocall connection failure: graceful [graceful]")
+	if(H == connected_holopad || H == calling_holopad)
+		if(!graceful)
+			calling_holopad.atom_say("Connection failure.")
+		qdel(src)
+		return
+
+	LAZYREMOVE(H.holo_calls, src)
+	dialed_holopads -= H
+	if(!dialed_holopads.len)
+		if(graceful)
+			calling_holopad.atom_say("Call rejected.")
+		log_debug("No recipients, terminating")
+		qdel(src)
+
+//Answers a call made to a holopad `H` which cannot be the calling holopad. Pads not in the call are ignored
+/datum/holocall/proc/Answer(obj/machinery/hologram/holopad/H)
+	testing("Holocall answer")
+	if(H == calling_holopad)
+		CRASH("How cute, a holopad tried to answer itself.")
+
+	if(!(H in dialed_holopads))
+		return
+
+	if(connected_holopad)
+		CRASH("Multi-connection holocall")
+
+	for(var/I in dialed_holopads)
+		if(I == H)
+			continue
+		Disconnect(I)
+
+	for(var/I in H.holo_calls)
+		var/datum/holocall/HC = I
+		if(HC != src)
+			HC.Disconnect(H)
+
+	connected_holopad = H
+
+	if(!Check())
+		return
+
+	hologram = H.activate_holo(user)
+	hologram.HC = src
+
+	//eyeobj code is horrid, this is the best copypasta I could make
+	eye = new
+	eye.origin = H
+	eye.eye_initialized = TRUE
+	eye.eye_user = user
+	eye.name = "Camera Eye ([user.name])"
+	user.remote_control = eye
+	user.reset_perspective(eye)
+	eye.setLoc(H.loc)
+
+//Checks the validity of a holocall and qdels itself if it's not. Returns TRUE if valid, FALSE otherwise
+/datum/holocall/proc/Check()
+	for(var/I in dialed_holopads)
+		var/obj/machinery/hologram/holopad/H = I
+		if(!(H.stat & NOPOWER))
```

Comment:
This will cause ConnectionFailure to be called whenever you dial a pad which DOES have power.

Just like that other comment this morning, remove the `!`